nifi-api/src/main/java/org/apache/nifi/processor/AbstractProcessor.java

```diff
@@ -27,7 +30,12 @@ public final void onTrigger(final ProcessContext context, final ProcessSessionFa
             onTrigger(context, session);
             session.commit();
         } catch (final Throwable t) {
-            getLogger().error("{} failed to process due to {}; rolling back session", new Object[]{this, t});
+            StringWriter stacktraceWriter = new StringWriter();
```

Comment:
Yep that would be the easiest way, and I can go ahead and throw this mess I added in above.